modules/benchmark.c

```diff
@@ -388,6 +388,27 @@ static void do_benchmark(void (*benchmark_function)(void), int entry)
        bench_image = icon_cache_get_image("benchmark.png");
        gtk_widget_show(bench_image);
 
+#if GTK_CHECK_VERSION(3, 0, 0)
+       GtkWidget *button;
+       GtkWidget *content_area;
+       GtkWidget *hbox;
+       GtkWidget *label;
+       
+       bench_dialog = gtk_dialog_new_with_buttons("",
+                                                  NULL,
```

Comment:
Shouldn't this second parameter be the parent window (`GTK_WINDOW(shell_get_main_shell()->window)`)? Otherwise, this dialog won't be modal.

If the GTK+3 code works in GTK+2, we can just get rid of both versions and keep the one that works in GTK+3.

Reply:
Well, GTK3 code doesn't work with GTK2 build, it has new warning when compiling:
```
/home/max/hardinfo/modules/benchmark.c: In function ‘do_benchmark’:
/home/max/hardinfo/modules/benchmark.c:438:13: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
        hbox = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 5);
             ^
```
And HardInfo crashes after choosing a benchmark. I think `gtk_box_new` doesn't work with GTK2, so it should be better to merge it "as is" with changing only `NULL` to `GTK_WINDOW(shell_get_main_shell()->window)` as you advised.

Reply:
@lpereira Please merge this PR with replacing `NULL` to `GTK_WINDOW(shell_get_main_shell()->window)`, it works both GTK2 and 3. It should be the best way, I guess.